crates/rolldown/src/stages/generate_stage/compute_cross_chunk_links.rs

```diff
@@ -234,47 +236,29 @@ impl<'a> GenerateStage<'a> {
       }
     }
 
-    let index_sorted_cross_chunk_imports = index_cross_chunk_imports
-      .into_iter()
-      // FIXME: Extra traversing. This is a workaround due to `par_bridge` doesn't ensure order https://github.com/rayon-rs/rayon/issues/551#issuecomment-882069261
-      .collect::<Vec<_>>()
-      .into_par_iter()
-      .map(|cross_chunk_imports| {
-        let mut cross_chunk_imports = cross_chunk_imports.into_iter().collect::<Vec<_>>();
-        cross_chunk_imports.sort_by_cached_key(|chunk_id| {
-          let mut resource_ids = chunk_graph.chunks[*chunk_id]
-            .modules
-            .iter()
-            .map(|id| {
-              self.link_output.module_table.normal_modules[*id].resource_id.expect_file().as_str()
-            })
-            .collect::<Vec<_>>();
-          resource_ids.sort_unstable();
-          resource_ids
-        });
-        cross_chunk_imports
-      })
-      .collect::<Vec<_>>();
-
     chunk_graph
       .chunks
       .iter_mut()
       .zip(
         imports_from_other_chunks_vec.into_iter().zip(
           chunk_meta_imports_from_external_modules_vec
             .into_iter()
-            .zip(index_sorted_cross_chunk_imports),
+            .zip(cross_chunk_imports.into_iter().zip(cross_chunk_dynamic_imports)),
         ),
       )
       .par_bridge()
       .for_each(
         |(
           chunk,
-          (imports_from_other_chunks, (imports_from_external_modules, cross_chunk_imports)),
+          (
+            imports_from_other_chunks,
+            (imports_from_external_modules, (cross_chunk_imports, cross_chunk_dynamic_imports)),
```

Comment:
Could you keep the `index` prefix above. The above `cross_chunk_imports: IndexVec<ChunkId, Vec<ChunkId>>` with this `cross_chunk_imports: Vec<ChunkId>>` could be really confusing,